sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -456,6 +456,40 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
   }
 
   /**
+   * Loads a `JavaRDD[String]` storing storing CSV rows and returns the result as a `DataFrame`.
+   *
+   * If the schema is not specified using `schema` function and `inferSchema` option is enabled,
+   * this function goes through the input once to determine the input schema.
+   *
+   * If the schema is not specified using `schema` function and `inferSchema` option is disabled,
+   * it determines the columns as string types and it reads only the first line to determine the
+   * names and the number of fields.
+   *
+   * @param csvRDD input RDD with one CSV row per record
+   * @since 2.2.0
+   */
+  @deprecated("Use csv(Dataset[String]) instead.", "2.2.0")
+  def csv(csvRDD: JavaRDD[String]): DataFrame = csv(csvRDD.rdd)
+
+  /**
+   * Loads a `RDD[String]` storing storing CSV rows and returns the result as a `DataFrame`.
+   *
+   * If the schema is not specified using `schema` function and `inferSchema` option is enabled,
+   * this function goes through the input once to determine the input schema.
+   *
+   * If the schema is not specified using `schema` function and `inferSchema` option is disabled,
+   * it determines the columns as string types and it reads only the first line to determine the
+   * names and the number of fields.
+   *
+   * @param csvRDD input RDD with one CSV row per record
+   * @since 2.2.0
+   */
+  @deprecated("Use csv(Dataset[String]) instead.", "2.2.0")
+  def csv(csvRDD: RDD[String]): DataFrame = {
```

Comment:
Wait ... I think we shouldn't introduce an RDD API in Scala side. I was thinking doing this within Python-side, or _maybe_ adding a private wrapper in Scala side if required .. Will takr a closer look tomorrow (KTS).

Reply:
Thanks for your reviewing :)
umm..  I followed `spark.read.json`'s way to add them. Although `json(jsonRDD :RDD[String]` has been deprecated, PySpark still use it to create a `DataFrame`. I think adding a private wrapper in Scala maybe better because not only PySpark but SparkR maybe need it.

Reply:
Yep. +1 for @HyukjinKwon's advice. We cannot add a **deprecated** method which doesn't exist in 2.2.0 at all.

Reply:
Yeah...It's weird to add a deprecated method. :) We either add a special wrapper for this purpose or doing this in python-side if possible and not complicated.